apps/els_lsp/src/els_methods.erl

```diff
@@ -449,7 +451,12 @@ workspace_executecommand(Params, State) ->
 %%==============================================================================
 
 -spec workspace_didchangewatchedfiles(map(), state()) -> result().
-workspace_didchangewatchedfiles(Params, State) ->
+workspace_didchangewatchedfiles(Params0, State) ->
+  #{open_buffers := OpenBuffers} = State,
+  #{<<"changes">> := Changes0} = Params0,
+  Changes = [C || #{<<"uri">> := Uri} = C <- Changes0,
+                  not lists:member(Uri, OpenBuffers)],
+  Params = Params0#{<<"changes">> => Changes},
```

Comment:
I agree, but then the provider would need access to the `State`. I may refactor the provider code later, since many things changed with the recent "collapse" into a single provider. For now, this should do.